### PR TITLE
feat(sagemaker): support dlc images in sagemaker model

### PR DESCRIPTION
**Pull request by vinayak-kukreja**
CDK currently does not support [deep learning container images for AWS Sagemaker models](https://docs.aws.amazon.com/sagemaker/latest/dg/pre-built-containers-frameworks-deep-learning.html). Looks like this functionality was proposed as part of a [PR](https://github.com/aws/aws-cdk/pull/17399/files#diff-356f35099770f68f4ceee2e63d34aad8729b0a9be6c933a0c05e999be7374685R98-R145) but did not make it to the RFC.

This PR is adding that functionality and tests for it.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-sagemaker/test/model.test.ts [hunk 1]
```diff
@@ -295,3 +295,39 @@ test('When importing a model by name, the ARN is constructed correctly', () => {
   // THEN
   expect(model.modelArn).toMatch(/arn:.+:sagemaker:us-west-2:123456789012:model\/my-name/);
 });
+
+test('sagemaker model with deep learning container images', () => {
```
Comment (corymhall) [inline packages/@aws-cdk/aws-sagemaker/test/model.test.ts:299]: Can you add a test that shows the mapping that will be created?

**Review by corymhall — CHANGES_REQUESTED** [review]
Just a couple of comments

File: packages/@aws-cdk/aws-sagemaker/lib/container-image.ts [hunk 2]
```diff
@@ -81,3 +90,57 @@ class AssetImage extends ContainerImage {
     };
   }
 }
+
+export class DlcEcrImage extends ContainerImage {
```
[the remark this answers is not in the document]
Comment (vinayak-kukreja) [reply packages/@aws-cdk/aws-sagemaker/lib/container-image.ts:94]: No, will remove this.

File: packages/@aws-cdk/aws-sagemaker/lib/container-image.ts [hunk 3]
```diff
@@ -81,3 +90,57 @@ class AssetImage extends ContainerImage {
...
+
+  constructor(private readonly repositoryName: string, private readonly tag: string, private readonly account?: string) {
+    super();
+  }
+
+  public bind(scope: Construct, model: Model): ContainerImageConfig {
+    const accountId = this.getAwsAccountId(scope);
```
[the remark this answers is not in the document]
Comment (vinayak-kukreja) [reply packages/@aws-cdk/aws-sagemaker/lib/container-image.ts:101]: Thanks Cory, this is great.

**Review by corymhall — APPROVED** [review]
🚀

**Pull request merged**